File: database/src/main/java/it/niedermann/nextcloud/deck/database/entity/AccountEntity.java
package it.niedermann.nextcloud.deck.database.entity;

import androidx.room.Entity;

import it.niedermann.nextcloud.deck.shared.model.Account;


@Entity(
        tableName = "Account",
        inheritSuperIndices = true,
        primaryKeys = {
                "id"
        },
        indices = {
        },
        foreignKeys = {
        }

)
public class AccountEntity extends Account {

    protected int id;
}

File: data/local/src/main/java/it/niedermann/nextcloud/deck/data/local/entity/AccountEntity.kt
package it.niedermann.nextcloud.deck.data.local.entity

import androidx.room3.Entity
import androidx.room3.PrimaryKey
import it.niedermann.nextcloud.deck.domain.model.Color
import it.niedermann.nextcloud.deck.domain.model.Version
import java.net.URL

@Entity(
    tableName = "Account",
)
@JvmRecord
data class AccountEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Long,
    val url: URL,
    val username: String,
    val token: String,
    val accountName: String,
    @androidx.room3.Embedded
    val capabilities: CapabilitiesEntity
)
